homeassistant/components/yolink/services.py

```diff
@@ -28,6 +29,13 @@ async def handle_speaker_hub_play_call(service_call: ServiceCall) -> None:
         device_registry = dr.async_get(hass)
         device_entry = device_registry.async_get(service_data[ATTR_TARGET_DEVICE])
         if device_entry is not None:
+            for entry_id in device_entry.config_entries:
+                if (entry := hass.config_entries.async_get_entry(entry_id)) is None:
+                    continue
+                if entry.domain == DOMAIN:
+                    break
+            if entry is None or entry.state == ConfigEntryState.NOT_LOADED:
+                raise ServiceValidationError("Config entry not found or not loaded!")
```

Comment:
A future improvement would be to add translation support for the exception message.